fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -0,0 +1,241 @@
+
+        final var rowMapper = new RowMapper<Map<String, Object>>() {
+            @Override
+            public Map<String, Object> mapRow(final ResultSet rs, final int rowNum) throws SQLException {
+                final Map<String, Object> map = new HashMap<>();
+                for (String f : fields) {
+                    var value = rs.getObject(f);
+                    if (value instanceof Timestamp) {
+                        value = ISO_INSTANT.format(Instant.ofEpochMilli(((Timestamp) value).getTime()));
+                    }
+                    map.put(f, value);
+                }
+                return map;
+            }
+        };
+
+        final List<Map<String, Object>> items = jdbcTemplate.query(sql.toString(), parameterSource, rowMapper);
+        final var pagination = new PaginationInfo(parameters.getMaxResults(), parameters.getOffset());
+        LOGGER.debug("Search query with parameters: {} - {}", sql, parameters);
+        return new SearchResult(items, pagination);
+    }
+
+    @Override
+    public void addUpdateIndex(final FedoraId fedoraId, final Instant created, final Instant modified,
+                               final Long size, final String mimetype) {
+        final var fullId = fedoraId.getFullId();
+        final var selectParams = new MapSqlParameterSource();
+        selectParams.addValue("fedora_id", fullId);
+        final var result = jdbcTemplate.queryForList("SELECT fedora_id FROM " + SIMPLE_SEARCH_TABLE +
+                        " WHERE fedora_id = :fedora_id",
+                selectParams);
+
+        final var txId = UUID.randomUUID().toString();
+        executeInDbTransaction(txId, status -> {
+            try {
+
+                final var params = new MapSqlParameterSource();
+                params.addValue("fedora_id", fullId);
+                params.addValue("modified", modified);
+
+                if (result.size() > 0) {
+                    //update
+                    final var sql = "UPDATE " + SIMPLE_SEARCH_TABLE +
+                            " SET modified=:modified  WHERE fedora_id = :fedora_id;";
+                    jdbcTemplate.update(sql, params);
+                } else {
+                    params.addValue("created", created);
+
+                    final var sql = "INSERT INTO " + SIMPLE_SEARCH_TABLE + " (fedora_id, modified, created) " +
+                            "VALUES(:fedora_id, :modified, :created)";
+                    jdbcTemplate.update(sql, params);
+                }
+                return null;
+            } catch (Exception e) {
+                status.setRollbackOnly();
+                throw new RepositoryRuntimeException("Failed add/updated the search index for : " + fullId, e);
+            }
+        });
+    }
+
+    @Override
+    public void removeFromIndex(final FedoraId fedoraId) {
+        final var txId = UUID.randomUUID().toString();
+        executeInDbTransaction(txId, status -> {
```

Comment:
I don't think this needs a transaction, does it?

Reply:
good catch.  I'll remove.